lib/StyleManager.vala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024 elementary, Inc. (https://elementary.io)
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+/**
+ * A class for managing the style of the application. This handles switching light and dark mode based
+ * based on system preference or application preference (see {@link color_scheme_override}), etc.
+ */
+public class Granite.StyleManager : Object {
+    private static Gtk.CssProvider? base_provider = null;
+    private static Gtk.CssProvider? dark_provider = null;
+    private static Gtk.CssProvider? app_provider = null;
+
+    private static GLib.Once<Granite.StyleManager> instance;
+    public static unowned Granite.StyleManager get_default () {
+        return instance.once (() => {
+            return new Granite.StyleManager ();
+        });
+    }
+
+    /**
+     * If this is set to NO_PREFERENCE the systems preferred color scheme will be used.
```

Comment:
I think we need to default to LIGHT if that's the case to preserve the current behavior. Otherwise I think it's technically a breaking change because developers might update and not expect that they need to support dark mode now by default

Reply:
Yeah that probably makes sense will update

Reply:
Actually thinking about it more, maybe we should go ahead and break and just bump everything to 8.0. I think we have to have a runtime break anyways because of the GNOME 47 release and we'll want to update the GNOME apps we ship